backend_jobs/recognition_pipeline/filters/filter_by_resolution.py

```diff
@@ -0,0 +1,55 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+from backend_jobs.recognition_pipeline.pipeline_lib.filter_by import FilterBy
+from backend_jobs.pipeline_utils.utils import get_provider
+from backend_jobs.ingestion_pipeline.main import IMAGE_PROVIDERS
+from backend_jobs.pipeline_utils import constants
+
+class FilterByResolution(FilterBy):
+    """ Checks if the image has a high enough resolution.
+
+    """
+
+    def is_supported(self, image):
+        """ Returns True iff image resolution is equal or bigger
+            than the minimum supported resolution supported by the provider.
+
+        """
+        image_attribute = image[constants.IMAGE_ATTRIBUTES][constants.RESOLUTION]
+        min_length = self.prerequisites['height']
+        min_width = self.prerequisites['width']
+        if image_attribute['width'] >= min_width and \
+          image_attribute['height'] >= min_length:
+            return True
+        return self.change_url_by_resolution(image)
```

Comment:
still, it mutates the dictionary that is being provided. 
Having side effects on a method parameter is usually a big red flag (it's often times the cause of various bugs). Especially if the method is not private (changes will leak to other libraries).


Can you think of an alternate approach?

Reply:
The method is private now. I can add another field instead of changing the current one/ return a new dict instead of mutating the current one (but I guess both those solutions don't really solve the problem). The thing is that it does not only need to return True if it support but actually return the url for the supported image. What do you think?

Reply:
returning a new dict technically is more correct, but is clunky (weird API signature).
What about we update this URL as another phase of the pipeline? 
\
Another approach would be to first have an stage in the pipeline which creates a 'url_for_vision_api' field (when possible), and then have a filter stage which removes entries which don't have this url set.